src/main/java/com/synopsys/integration/alert/component/settings/PasswordResetService.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (C) 2019 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.component.settings;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.channel.email.EmailChannel;
+import com.synopsys.integration.alert.channel.email.EmailMessagingService;
+import com.synopsys.integration.alert.channel.email.EmailProperties;
+import com.synopsys.integration.alert.channel.email.template.EmailTarget;
+import com.synopsys.integration.alert.common.AlertProperties;
+import com.synopsys.integration.alert.common.configuration.FieldAccessor;
+import com.synopsys.integration.alert.common.database.BaseConfigurationAccessor;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.EmailPropertyKeys;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.database.api.configuration.model.ConfigurationModel;
+import com.synopsys.integration.alert.database.api.user.UserAccessor;
+import com.synopsys.integration.alert.database.api.user.UserModel;
+
+@Component
+public class PasswordResetService {
+    private static final int TEMP_PASSWORD_LENGTH = 50;
```

Comment:
50 seems a bit long. Do we think 10 would suffice as well? I know sometimes I don't have the opportunity to copy and paste when changing passwords like this and have to type in the new value manually.

Reply:
Luckily we allow users to copy and paste the password since we are the application developers ;)

Reply:
Since our code is open source, and we do not currently limit login attempts, the ability to brute force a login is significantly stymied by increasing the number of characters in the temp password.
36^10 ≈ 3.65 x 10^15
36^50 ≈ 6.53 x 10^77